app/models/concerns/spotlight/browse_category_search_builder.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Spotlight
+  # Enforce exhibit visibility for index queries
+  module BrowseCategorySearchBuilder
+    extend ActiveSupport::Concern
+
+    included do
+      self.default_processor_chain = %i[restrict_to_browse_category] + default_processor_chain
+    end
+
+    # Adds a filter that excludes resources that have been marked as not-visible
+    def restrict_to_browse_category(solr_params)
+      return unless current_exhibit && blacklight_params[:browse_category_id].present?
+
+      search = current_exhibit.searches.find(blacklight_params[:browse_category_id])
+      search_builder = self.class.new(@processor_chain, @scope)
+      browse_category_search_params = search_builder.with(search.query_params).to_hash
+
+      # something clever with q
```

Comment:
For our sanity, I think the "something clever" needs to be an extension on https://github.com/projectblacklight/blacklight/pull/2434.

Previously, it looks like we just concatenated the two query strings together and called it a day, which probably caused some weird behavior wrt Solr's `mm` making some of the browse category tokens optional. 🤷‍♂️